include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -1934,6 +1934,15 @@ MatrixFree<dim, Number, VectorizedArrayType>::initialize_indices(
                                 di.vector_partitioner_face_variants[4],
                                 loop_over_all_faces);
             }
+          else
+            {
+              di.vector_partitioner_face_variants[3].reset(
+                new Utilities::MPI::Partitioner(part.locally_owned_range(),
+                                                part.get_mpi_communicator()));
```

Comment:
We've tried to avoid bare `new` operations recently. It is easier to do
```
di.vector_partitioner_face_variants[3]
  = std::make_shared<Utilities::MPI::Partitioner>(part.locally_owned_range(),
                              ￼                                                part.get_mpi_communicator());
```
and the same for the next line.

Reply:
@bangerth Should I replace also the other `new` in that file?

Reply:
@bangerth I would suggest, we keep it this way and I replace the `new`s in the `matrixfree` folder in a follow-up PR. Is that a deal?

Reply:
Yes, I hadn't realized that there were others in the same file. So yes, let's merge this once the tester is through and I'll let you do the news in a separate PR if you're willing to do that.